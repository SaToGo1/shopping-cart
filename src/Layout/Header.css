header {
    padding: 10px;
    display: flex;
    align-items: center;

    gap: 5%;
    margin-right: max(7%, 50px);
    margin-left: 50px;

    min-height: 10vh;
}

.header__MainPageDiv {

}
    .header__h1 {
        font-size: 4rem;
    }

.header__ShopDiv {
    margin-left: auto;
}

.header__CartDiv {

}
    .header__p {
        font-size: 3rem;
    }


.header__button {
    color: var(--letter);
    background-color: var(--main);
    border: none;
    padding: 5px 10px;
}

    .header__button:hover {
        cursor: pointer;

        outline: 1px solid var(--letter);
        border-radius: 3px;
    }

    .header__button:active {
        cursor: pointer;
        color: var(--letter2);
        outline: 1px solid var(--letter2);
    }