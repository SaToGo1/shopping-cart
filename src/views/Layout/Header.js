import './Header.css'
import { Link } from 'react-router-dom'

const Header = ({ handleCartDisplay, isCartVisible }) => {

    return (
        <header className={(isCartVisible? 'header__cart-open' : '')}>
            <section>
                <Link to='/'>
                    <button className='header__button'>
                        <h1 className='header__h1'>Main Page</h1>
                    </button>
                </Link>
            </section>

            <section className='header__ShopDiv'>
                <Link to='/shop'>
                    <button className='header__button'>
                        <p className='header__p'>Shop</p>
                    </button>
                </Link>
            </section>
            <section>
                <button className='header__button' onClick={handleCartDisplay}>
                    <p className='header__p'>Cart</p>
                </button>
            </section>
        </header>
    )
}

export default Header;