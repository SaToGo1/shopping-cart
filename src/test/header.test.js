import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Header from '../components/header';

describe('The Header ', () => {

  it('renders header Title', () => {
    render(<Header />);
    const title = screen.getByText(/Main Page/i);
    expect(title).toBeInTheDocument();
  });

  it('renders header Shop', () => {
    render(<Header />);
    const title = screen.getByText(/Shop/i);
    expect(title).toBeInTheDocument();
  });

  it('renders header Cart', () => {
    render(<Header />);
    const title = screen.getByText(/Cart/i);
    expect(title).toBeInTheDocument();
  });

  it('matches Snapshot', () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
})