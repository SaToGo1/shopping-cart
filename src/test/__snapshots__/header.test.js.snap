// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Header  matches Snapshot 1`] = `
<div>
  <header>
    <div
      class="header__MainPageDiv"
    >
      <button
        class="header__button"
      >
        <h1
          class="header__h1"
        >
          Main Page
        </h1>
      </button>
    </div>
    <div
      class="header__ShopDiv"
    >
      <button
        class="header__button"
      >
        <p
          class="header__p"
        >
          Shop
        </p>
      </button>
    </div>
    <div
      class="header__CartDiv"
    >
      <button
        class="header__button"
      >
        <p
          class="header__p"
        >
          Cart
        </p>
      </button>
    </div>
  </header>
</div>
`;
